src/scipp/core/reduction.py

```diff
@@ -358,3 +847,111 @@ def any(x: VariableLikeType, dim: Optional[str] = None) -> VariableLikeType:
         return _call_cpp_func(_cpp.any, x)
     else:
         return _call_cpp_func(_cpp.any, x, dim=dim)
+
+
+def _reduce_with_numpy(
+    x: VariableLikeType,
+    *,
+    dim: Dims = None,
+    sc_func: Callable[..., VariableLikeType],
+    np_func: Callable[..., np.ndarray],
+    np_ma_func: Callable[..., np.ndarray],
+    unit_func: Callable[[_cpp.Unit], _cpp.Unit],
+    kwargs: dict[str, Any],
+) -> VariableLikeType:
+    if isinstance(x, _cpp.Dataset):
+        return _cpp.Dataset(
+            {
+                k: _reduce_with_numpy(
+                    v,
+                    dim=dim,
+                    sc_func=sc_func,
+                    np_func=np_func,
+                    np_ma_func=np_ma_func,
+                    unit_func=unit_func,
+                    kwargs=kwargs,
+                )
+                for k, v in x.items()
+            }
+        )
+    if isinstance(x, DataGroup):
+        return data_group_nary(sc_func, x, dim=dim, **kwargs)
+
+    _expect_no_variance(x, sc_func.__name__)
+    reduced_dims, out_dims, axis = _split_dims(x, dim)
+    if isinstance(x, _cpp.Variable):
+        return array(
+            dims=out_dims,
+            values=np_func(x.values, axis=axis, **kwargs),
+            unit=unit_func(x.unit),
+        )
+    if isinstance(x, _cpp.DataArray):
```

Comment:
Can you check if this is the same as https://github.com/scipp/scipp/blob/8e19ec99ffc15cfacee01927b703b17e9a219e75/src/scipp/core/concepts.py#L33-L41 and https://github.com/scipp/scipp/blob/8e19ec99ffc15cfacee01927b703b17e9a219e75/src/scipp/core/concepts.py#L75-L81. I think there must be at least some of `scipp.core.concepts` that you can reuse.

Reply:
Right, I didn't remember that this file exists.